Framework/Algorithms/src/MonteCarloAbsorption.cpp

```diff
@@ -101,11 +101,18 @@ void MonteCarloAbsorption::init() {
   declareProperty(std::make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
                                                         Direction::Output),
                   "The name to use for the output workspace.");
+
+  declareProperty("SimulateTracksForEachWavelength", false,
```

Comment:
I think this property should be added in the end to avoid breaking the API when used by positional arguments. Also I am not sure about the choice of the default, since that also changes the results.

Reply:
I think there's a trade off here - I could move the parameter to the end but the new parameter is associated with the "NumberOfWavelengthPoints" parameter eg NumberOfWavelengthPoints is disabled if SimulateTracksForEachWavelength is False. So it will be less easy for users to understand what the parameters are doing especially for users running the algorithm from the UI

The reason for the selecting SimulateTracksForEachWavelength=False as the default is that the current logic for generating random tracks and a scatter point has no dependence on the wavelength at the moment. So there's no benefit at the moment in repeating the track simulation for different wavelengths. It just makes the calculation run slower. So false is the recommended value. You're right - this does change the results slightly, particularly if the simulation isn't converged. That's why I've had to change a lot of reference files in the unit\system tests.

Reply:
I was wondering if you would set `SimulateTracksForEachWavelenght=True` in the tests, do you obtain the exact same results or there are little differences still due to different sampling of random numbers?

Reply:
I've switched the order of two nested loops in the code (the events loop and the wavelength loop) so you would get slightly different results with SimulateTracksForEachWavelength=True beacuse the random number sequences are different. I had to do this to allow the old\new approaches to coexist in the code.

Apart from the random number differences, SimulateTracksForEachWavelength=True does allow the original process\calculation to be performed.

Reply:
I understand the tradeoff, and the breaking change seems justified, however this needs to be clearly stated in the release notes at least. Also I think renaming the property to `ReSimulateTracksForEachWavelength` will help the clarity.

Reply:
Yes will do. I wasn't 100% happy with the name of the new parameter so that will make things clearer. Perhaps ReSimulateTracksForDifferentWavelengths is even better? The word "each" makes it sound like resimulation will occur for all wavelength bins which isn't true if you are setting NumberOfWavelengthPoints to 10 (for example)

Reply:
Yes, that's even better.